### PR TITLE
feat: Enhance security bootstrap to support Postgres

### LINKED ISSUE #4836: Enhance Security Bootstrap to support PostgresSQL
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for all EdgeX services will use the PostgreSQL database when security services are enabled.

### Description [**REQUIRED**]
 Since PostgreSQL will be added as one of the EdgeX Foundry's implementation databases, Security Bootstrapper, Security SecretStore and other security services need to be enhanced for securing access to PostgreSQL database.

### Describe the solution you'd like
 New entrypoint script and security bootstrap handler will be needed for securing access to PostgreSQL.

### PR DESCRIPTION
**Pull request by lindseysimple**
Resolves #4836. Enhance security bootstrap/secret-store to support Postgres.

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: cmd/security-bootstrapper/res/configuration.yaml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 LogLevel: INFO
+Database:
+  Type: postgres
```
Comment (cloudxxx8) [inline cmd/security-bootstrapper/res/configuration.yaml:3]: remove this config
just reuse the Database config under StageGate
The value should be replaced by the environment variables

File: cmd/security-secretstore-setup/res/configuration.yaml [hunk 2]
```diff
@@ -38,6 +38,8 @@ SecretStore:
   PasswordProviderArgs: []
   RevokeRootTokens: true
   ConsulSecretsAdminTokenPath: /tmp/edgex/secrets/edgex-consul/admin/token.json
+Database:
+  Type: "postgres"
```
Comment (cloudxxx8) [inline cmd/security-secretstore-setup/res/configuration.yaml:42]: move `Type` under the exsting `Database` section, and the default value is still redisdb

File: internal/security/secretstore/client/secretclient.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+type SecretClient struct {
+	client               internal.HttpCaller
+	rootToken            string
+	secretServiceBaseURL string
+	loggingClient        logger.LoggingClient
+}
+
+func NewSecretClient(
```
Comment (cloudxxx8) [inline internal/security/secretstore/client/secretclient.go:38]: are we able to reuse the secret client from go-mod-secret?

File: internal/security/secretstore/init.go [hunk 4]
```diff
@@ -74,7 +78,7 @@ func NewBootstrap(insecureSkipVerify bool, vaultInterval int) *Bootstrap {
 	return &Bootstrap{
 		insecureSkipVerify: insecureSkipVerify,
 		vaultInterval:      vaultInterval,
-		validKnownSecrets:  map[string]bool{redisSecretName: true, messagebusSecretName: true},
+		validKnownSecrets:  map[string]bool{redisSecretName: true, postgresSecretName: true, messagebusSecretName: true},
 	}
```
Comment (cloudxxx8) [inline internal/security/secretstore/init.go:82]: check the database type from the config first, and add valid known secret for either redisSecretName or postgresSecretName

File: internal/security/secretstore/tls/cert.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+package tls
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func createCA(commonName, certPath, keyPath string) error {
+	caSubject := fmt.Sprintf("/C=UK/O=IOTech/CN=%s", commonName)
```
Comment (cloudxxx8) [inline internal/security/secretstore/tls/cert.go:15]:
```suggestion
	caSubject := fmt.Sprintf("/C=UK/O=EdgeXFoundry/CN=%s", commonName)
```
use a neutral name

File: internal/security/secretstore/tls/cert.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+package tls
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func createCA(commonName, certPath, keyPath string) error {
```
Comment (cloudxxx8) [inline internal/security/secretstore/tls/cert.go:14]: Allow users to pass all the information, and use default velues if they are empty

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: cmd/security-bootstrapper/res/configuration.yaml (hunk 1 above)
Comment (lindseysimple) [reply cmd/security-bootstrapper/res/configuration.yaml]: Done

File: cmd/security-secretstore-setup/res/configuration.yaml (hunk 2 above)
Comment (lindseysimple) [reply cmd/security-secretstore-setup/res/configuration.yaml]: Done

File: internal/security/secretstore/client/secretclient.go (hunk 3 above)
Comment (lindseysimple) [reply internal/security/secretstore/client/secretclient.go]: Removed all tls package related files in this PR, will open another issue to support Postgres TLS connection.

File: internal/security/secretstore/init.go (hunk 4 above)
Comment (lindseysimple) [reply internal/security/secretstore/init.go]: Updated to only add `postgresSecretName` if database type is postgres.

File: internal/security/secretstore/tls/cert.go (hunk 5 above)
Comment (lindseysimple) [reply internal/security/secretstore/tls/cert.go]: tls related comments will be addressed in the following PR.

File: internal/security/secretstore/init.go [hunk 7]
```diff
@@ -53,6 +56,7 @@
 const (
 	addKnownSecretsEnv   = "EDGEX_ADD_KNOWN_SECRETS" // nolint:gosec
 	redisSecretName      = "redisdb"
+	postgresSecretName   = "postgres"
```
[the remark this answers is not in the document]
Comment (lindseysimple) [reply internal/security/secretstore/init.go:59]: `postgresSecretName` indicates the secret path on the secret store, which aligns with the existing `redisSecretName` path.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**